### PR TITLE
added resizing functionality to the area rects

### PR DESCRIPTION
**Pull request by valeriegc**

### CODE REVIEW

File: src/routes/Canvas.svelte [hunk 1]
```diff
@@ -45,6 +38,8 @@
 	}
 </script>
 
+<div class="lol" use:resizeArea />
+
```
Comment (AlexanderKolberg) [inline src/routes/Canvas.svelte]: What is this?

File: src/actions/Resize.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	function whileMoving(event) {
+		if (!moving) return;
+
+		let delta;
+
+		delta = firstPosition.x - event.pageX;
+		element.style.width = `${firstRect.width - delta}px`;
+		delta = event.pageY - firstPosition.y;
+		element.style.height = `${firstRect.height + delta}px`;
```
Comment (AlexanderKolberg) [inline src/actions/Resize.ts]:
```suggestion
		const deltaX = firstPosition.x - event.pageX;
		element.style.width = `${firstRect.width - deltaX}px`;
		
		const deltaY = event.pageY - firstPosition.y;
		element.style.height = `${firstRect.height + deltaY}px`;
```

File: src/actions/Resize.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			left: rect.left - parent.left,
+			right: parent.right - rect.right,
+			top: rect.top - parent.top,
+			bottom: parent.bottom - rect.bottom
+		};
+
+		firstPosition = { x: event.pageX, y: event.pageY };
+		moving.classList.add('selected');
+	}
+	function onMouseup() {
```
Comment (AlexanderKolberg) [inline src/actions/Resize.ts:35]:
```suggestion

	function onMouseup() {
```

File: src/actions/Resize.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+export function resizeArea(element) {
```
Comment (AlexanderKolberg) [inline src/actions/Resize.ts]: The whole file looks like its missing types?

**Review by AlexanderKolberg — CHANGES_REQUESTED** [review]

File: src/routes/Canvas.svelte (hunk 1 above)
Comment (valeriegc) [reply src/routes/Canvas.svelte]: removed

File: src/actions/Resize.ts (hunk 2 above)
Comment (valeriegc) [reply src/actions/Resize.ts]: thanks! fixed.

File: src/actions/Resize.ts (hunk 3 above)
Comment (valeriegc) [reply src/actions/Resize.ts:35]: according to your words "i think i am too pedantic"

File: src/routes/Canvas.svelte [hunk 5]
```diff
@@ -59,6 +52,7 @@
 			{#each $mapObjects as _, i}
 				<div
 					class="areaRect"
+					use:resizeArea={$mapObjects}
```
Comment (AlexanderKolberg) [inline src/routes/Canvas.svelte]: Why are you feeding the mapsObjects?

**Review by AlexanderKolberg — APPROVED** [review]

**Pull request merged**